### PR TITLE
MAT-3789, MAT-6153

### PR DESCRIPTION
**Pull request by mcmcphillips**
## MADiE PR

Jira Ticket: [MAT-3789](https://jira.cms.gov/browse/MAT-3789)
Jira Ticket: [MAT-6153](https://jira.cms.gov/browse/MAT-6153)

(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/components/editMeasure/details/MeasureReferences/MeasureReferences.tsx [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+              placeHolder={{ name: "Select", value: "" }}
+              inputProps={{
+                "data-testid": `measure-referenceType-input`,
+              }}
+              data-testid={`measure-referenceType`}
+              disabled={!canEdit}
+              required
+              //   SelectDisplayProps={{
+              //     "aria-required": "true",
+              //   }}
```
Comment (adongare) [inline src/components/editMeasure/details/MeasureReferences/MeasureReferences.tsx:364]: if not required, can be removed.

File: src/components/editMeasure/details/MeasureReferences/MeasureReferences.tsx [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+              //    value={selectedValueSet ? selectedValueSet.oid : ""}
+              //    renderValue={(value) => {
+              //      if (value === "") {
+              //        return placeHolder("Select Value Set / Direct Reference Code");
+              //      }
+              //      return selectedValueSet?.display_name;
+              //    }}
+
+              // onChange={handleValueSetChange}
+            />
```
Comment (adongare) [inline src/components/editMeasure/details/MeasureReferences/MeasureReferences.tsx]: can be removed if not required anymore

File: src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitions.tsx [hunk 3]
```diff
@@ -249,8 +249,9 @@ const MeasureDefinitions = (props) => {
             <tbody data-testId="measure-definitions-table-body">
               {visibleDefinitions?.length > 0 ? (
                 visibleDefinitions.map((measureDefinition, index) => (
-                  <MeasureDefinitionRow
-                    measureDefinition={measureDefinition}
+                  <MeasureMetaDataRow
+                    value={measureDefinition}
+                    type="definition"
```
Comment (adongare) [inline src/components/editMeasure/details/MeasureDefinitions/MeasureDefinitions.tsx]: only if it makes sense else please ignore it.
```suggestion
                    name={measureDefinition.term}
                    description={measureDefinition.definition}
```

File: src/components/editMeasure/details/MeasureMetaDataRow.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    return (
+      <tr>
+        <td>{referenceType}</td>
+        <td>{referenceText}</td>
+      </tr>
+    );
+  }
+};
+
+export default MeasureMetaDataRow;
```
Comment (adongare) [inline src/components/editMeasure/details/MeasureMetaDataRow.tsx]:
```suggestion
const MeasureMetaDataRow = ({ name, description }) => {
   return (
      <tr>
        <td>{name}</td>
        <td>{description}</td>
      </tr>
    );
};

export default MeasureMetaDataRow;
```

File: src/components/editMeasure/details/MeasureReferences/MeasureReferences.tsx [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+                </th>
+              </tr>
+            </thead>
+            <tbody data-testId="measure-references-table-body">
+              {visibleReferences?.length > 0 ? (
+                visibleReferences.map((reference) => (
+                  <MeasureMetaDataRow
+                    value={reference}
+                    key={reference.id}
+                    type="reference"
```
Comment (adongare) [inline src/components/editMeasure/details/MeasureReferences/MeasureReferences.tsx]:
```suggestion
                    name={reference.referenceType}
                    description={reference.referenceText}
                    key={reference.id}
```

File: src/components/editMeasure/details/MeasureReferences/MeasureReferences.tsx (hunk 1 above)
Comment (adongare) [inline src/components/editMeasure/details/MeasureReferences/MeasureReferences.tsx]: if not required, can be removed.

File: src/components/editMeasure/details/MeasureReferences/MeasureReferencesValidator.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import * as Yup from "yup";
+
+export const MeasureReferencesValidator = Yup.object().shape({
+  referenceType: Yup.string().required("Measure Reference Type is required."),
+  referenceText: Yup.string()
+    .required("Measure References is required.")
```
Comment (adongare) [inline src/components/editMeasure/details/MeasureReferences/MeasureReferencesValidator.ts:6]: not sure if this is singular or plural

File: src/components/editMeasure/details/MeasureReferences/MeasureReferencesValidator.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+    .max(250, "Measure References cannot be more than 250 characters."),
```
Comment (adongare) [inline src/components/editMeasure/details/MeasureReferences/MeasureReferencesValidator.ts]: not sure if this should be singular or plural: References

File: src/components/editMeasure/details/MeasureReferences/MeasureReferences.tsx [hunk 8]
```diff
@@ -0,0 +1,404 @@
...
+        }
+        if (type1 === type2) {
+          if (reference1 < reference2) {
+            return -1;
+          }
+          if (reference2 > reference1) {
+            return 1;
+          }
+        }
+        return 0;
```
Comment (adongare) [inline src/components/editMeasure/details/MeasureReferences/MeasureReferences.tsx:109]: same but lil shorter
```suggestion
  const typeComparison = type1.localeCompare(type2);
  if (typeComparison === 0) {
    return reference1.localeCompare(reference2);
  }
  return typeComparison;  
```

File: src/components/editMeasure/details/MeasureReferences/MeasureReferences.tsx (hunk 2 above)
Comment (mcmcphillips) [reply src/components/editMeasure/details/MeasureReferences/MeasureReferences.tsx]: oh man

**Review by adongare — APPROVED** [review]

**Pull request merged**